meilisearch-http/src/error.rs

```diff
@@ -188,6 +202,7 @@ impl fmt::Display for Error {
             Self::CreateIndex(err) => write!(f, "Impossible to create index; {}", err),
             Self::DocumentNotFound(document_id) => write!(f, "Document with id {} not found", document_id),
             Self::IndexNotFound(index_uid) => write!(f, "Index {} not found", index_uid),
+            Self::IndexAlreadyExists(index_uid) => write!(f, "Index {} already exists", index_uid),
```

Comment:
@Kerollmops `IndexAlreadyExists` exist in `meilisearch_core::Error` and `meilisearch_http::ResponseError` but not in `meilisearch_http::Error`. Before `IndexAlreadyExists` was directly converted from `meilisearch_core::Error` to `meilisearch_http::ResponseError` because http route was directly returning errors to client. Now we have an intermediate function that returns `meilisearch_http::Error` which seem to be a more generalistic error

Reply:
Ok, seems fair!